ThunkLibs/include/common/Host.h

```diff
@@ -86,3 +59,20 @@ struct fex_guest_function_ptr {
     init_fn (); \
   }
 
+struct GuestcallInfo {
+  uintptr_t HostPacker;
+  void (*CallCallback)(uintptr_t GuestUnpacker, uintptr_t GuestTarget, void* argsrv);
+  uintptr_t GuestUnpacker;
+  uintptr_t GuestTarget;
+};
+
+// Helper macro for reading an internal argument passed through the `r11`
+// host register. This macro must be placed at the very beginning of
+// the function it is used in.
+#if defined(_M_X86_64)
+#define LOAD_INTERNAL_GUESTPTR_ARG(target_variable) \
+  asm("mov %%r11, %0" : "=r" (target_variable))
```

Comment:
these needs to be `volatile` to prevent the compiler reordering them

Reply:
Also, I'd rather name the define to something that includes "custom abi" to make it clear what is happening there.

Reply:
(sidenote: We also need volatile on the guest side as well)